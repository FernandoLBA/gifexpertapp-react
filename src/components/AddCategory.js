import React, { useState } from 'react';
import PropTypes from 'prop-types'


export const AddCategory = ( { setCategories } ) => {

     // Creamos un hook
     const [ inputValue, setInputValue ] = useState( '' );

     const handleInputChange = ( e ) => {
          setInputValue( e.target.value );
     };

     const handleSubmit = ( e ) => {
          e.preventDefault();

          if ( inputValue.trim().length > 1 ) {
               // coloca el inputValue primero
               setCategories( categories => [ inputValue, ...categories ] );
               // limpia borra el input
               setInputValue( '' );
          };
     };

     return (
               <form onSubmit={ handleSubmit }>
                    {/* Este párrafo se creó para pasar una prueba por eso no es necesario que se visualice */}
                    <p style={{display:'none' }}>{ inputValue }</p>
                    
                    <input
                         type="text"
                         value={ inputValue }
                         onChange={ handleInputChange }
                         placeholder="Busque aquí..."
                    />
               </form>
     );
};

AddCategory.propTypes = {
     setCategories: PropTypes.func.isRequired
};