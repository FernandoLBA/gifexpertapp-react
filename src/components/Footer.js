import React from 'react'

const Footer = () => {

     const year = new Date().getFullYear();

     return (
          <footer>
               <p>Desarrollada por: Fernando Barrios { year }.</p>
          </footer>
     )
}

export default Footer
