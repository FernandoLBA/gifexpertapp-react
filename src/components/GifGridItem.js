import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ( { title, url } ) => {

     return (
          <div className="card animate__animated animate__zoomIn">
              <img src={ url } alt={ title }></img>
              {/* limita el contenido del texto */}
              {/* <p>{ title.substr( 0, 15 ) }...</p> */}
              <p>{ title }</p>
          </div>
     );
};

export default GifGridItem;

GifGridItem.propTypes = {
     title: PropTypes.string.isRequired,
     url: PropTypes.string.isRequired
};
