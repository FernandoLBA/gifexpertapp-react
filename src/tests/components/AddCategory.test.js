import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';
import '@testing-library/jest-dom';


describe( 'Pruebas al componente <AddCategory />', () => {
     
     // creamos una función así:
     // const setCategories = () => {};

     // o una función así:
     const setCategories = jest.fn();

     // le mandamos la función requerida al componente
     let wrapper = shallow( <AddCategory setCategories={ setCategories } /> );

     // Reinicia todo
     beforeEach( () => {
          // Esto reinicializará el jest
          jest.clearAllMocks();

          // reestablecerá el componente
          wrapper = shallow( <AddCategory setCategories={ setCategories } /> );
     } );

     test( 'Debe mostrar el componente <AddCategory /> correctamente', () => {
          expect( wrapper ).toMatchSnapshot();
     } );

     test( 'Debe cambiar la caja de texto', () => {
          const input = wrapper.find( 'input' );
          const value = 'Hola Mundo';

          input.simulate( 'change', { target: { value } } );

          expect( wrapper.find( 'p' ).text().trim() ).toBe( value );
     } );

     test( 'No debe postear la información con submit', () => {
          
          // le mandamos el evento al submit
          wrapper.find( 'form' ).simulate( 'submit', { preventDefault(){} } );

          expect( setCategories ).not.toHaveBeenCalled();
     } );

     test('Debe llamar setCategories y limpiar la caja de texto', () => {
          
          const input = wrapper.find( 'input' );
          const value = 'Hola Mundo';
          
          // simular inputChange
          input.simulate( 'change', { target: { value } } );

          expect( wrapper.find('p').text().trim() ).toBe( value );

          // Simular submit
          wrapper.find( 'form' ).simulate( 'submit', { preventDefault(){} } );

          // Se debe llamar a setCategories
          expect( setCategories ).toHaveBeenCalled();

          // El input debe estar vacío
          expect( input.text() ).toBe( '' );
     });
} );
