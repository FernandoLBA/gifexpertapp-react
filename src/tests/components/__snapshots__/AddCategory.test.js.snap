// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <AddCategory /> Debe mostrar el componente <AddCategory /> correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p
    style={
      Object {
        "display": "none",
      }
    }
  />
  <input
    onChange={[Function]}
    placeholder="Busque aquÃ­..."
    type="text"
    value=""
  />
</form>
`;
